substrate/frame/system/src/offchain.rs

```diff
@@ -457,38 +447,54 @@ pub trait SigningTypes: crate::Config {
 	type Signature: Clone + PartialEq + core::fmt::Debug + codec::Codec + scale_info::TypeInfo;
 }
 
-/// A definition of types required to submit transactions from within the runtime.
-pub trait SendTransactionTypes<LocalCall> {
-	/// The extrinsic type expected by the runtime.
-	type Extrinsic: ExtrinsicT<Call = Self::OverarchingCall> + codec::Encode;
+/// Common interface for the `CreateTransaction` trait family to unify the `Call` type.
+pub trait CreateTransactionBase<LocalCall> {
+	/// The extrinsic.
+	type Extrinsic: ExtrinsicLike + Encode;
+
 	/// The runtime's call type.
 	///
 	/// This has additional bound to be able to be created from pallet-local `Call` types.
-	type OverarchingCall: From<LocalCall> + codec::Encode;
+	type RuntimeCall: From<LocalCall> + Encode;
 }
 
-/// Create signed transaction.
-///
-/// This trait is meant to be implemented by the runtime and is responsible for constructing
-/// a payload to be signed and contained within the extrinsic.
-/// This will most likely include creation of `SignedExtra` (a set of `SignedExtensions`).
-/// Note that the result can be altered by inspecting the `Call` (for instance adjusting
-/// fees, or mortality depending on the `pallet` being called).
+/// Interface for creating a transaction.
+pub trait CreateTransaction<LocalCall>: CreateTransactionBase<LocalCall> {
+	/// The extension.
+	type Extension: TypeInfo;
+
+	/// Create a transaction using the call and the desired transaction extension.
+	fn create_transaction(
+		call: <Self as CreateTransactionBase<LocalCall>>::RuntimeCall,
+		extension: Self::Extension,
+	) -> Self::Extrinsic;
+}
+
+/// Interface for creating an old-school signed transaction.
 pub trait CreateSignedTransaction<LocalCall>:
-	SendTransactionTypes<LocalCall> + SigningTypes
+	CreateTransactionBase<LocalCall> + SigningTypes
 {
+	/// TODO: revisit this, currently here for metadata purposes.
+	type SignaturePayload: SignaturePayload;
```

Comment:
I don't see it being used outside tests in frame-support, do I miss something?

Reply:
let's see: https://github.com/paritytech/polkadot-sdk/pull/5440

Reply:
Seems there is an error on the CI on that change.

Reply:
I'm not sure, it seems to be the same tests as this PR, but we can see at the end

Reply:
Merged.